File: Algorithm/src/Algorithm/Python/backjun/stack/J9012.py
for i in range(int(input())):
    s = input()
    stack = []
    ck = True
    for c in s:
        if c == '(':
            stack.append(c)
        else:
            if len(stack) > 0:
                stack.pop()
            else:
                ck = False
                break
    if len(stack) > 0:
        ck = False
    # if ck:
    #     print("YES")
    # else:
    #     print("NO")
    print("YES") if ck else print("NO")

File: Algorithm/src/Algorithm/Java/backjun/Stack/J9012.java
package Algorithm.Java.backjun.Stack;

import java.util.*;

public class J9012 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        while (n --> 0) {
            String s = sc.next();
            Stack<Character> stk = new Stack<>();
            boolean ck = true;
            char[] arr = s.toCharArray();
            for(char c : arr) {
                if (c == '(')
                    stk.push(c);
                else {
                    if(!stk.isEmpty())
                        stk.pop();
                    else {
                        ck = false;
                        break;
                    }
                }
            }
            if (!stk.isEmpty()) {
                ck = false;
            }
            System.out.println(ck ? "YES" : "NO");
        }
    }
}
